src/factory.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * factory.ts
+ * Generate a factory graph using a recursive algorithm.
+ * lgfrbcsgo & Nikolaus - October 2020
+ */
+
+import { items, Item, Ore, isOre } from "./items"
+import { findRecipe } from "./recipes"
+import { ContainerNode, FactoryGraph, IndustryNode, PerMinute } from "./graph"
+
+/**
+ * Add to the a factory graph all nodes required to produce and store a given item
+ * produced at a given rate. Recursively call this function to produce all ingredients.
+ * @param factory FactoryGraph to be modified
+ * @param item Item to be built
+ * @param rate Rate at which to build the product
+ */
+export function buildDependencies(
+    factory: FactoryGraph,
+    item: Item,
+    rate: PerMinute,
+): ContainerNode {
+    /** Get containers holding this product */
+    const containers = Array.from(factory.getContainers(item))
+    /** If ingredient is an ore, link from new or existing ore container */
+    if (isOre(item)) {
+        let container: ContainerNode | undefined = undefined
+        if (containers.length > 0) {
+            /*  Use existing ore container */
+            container = containers[0]
```

Comment:
Is it intentional to ignore the link limit here?

Reply:
Nice find. Fixed in https://github.com/lgfrbcsgo/du-factory-generator/commit/f6aa6bd0ac46c5f55bfc3d0c40b66df966be277d